src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3820,33 +3755,19 @@ private static void EmitTimeoutCheck(IndentedTextWriter writer, bool hasTimeout)
             }
         }
 
-        private static bool EmitInitializeCultureForTryMatchAtCurrentPositionIfNecessary(IndentedTextWriter writer, RegexMethod rm, AnalysisResults analysis)
-        {
-            if (analysis.HasIgnoreCase && ((RegexOptions)rm.Options & RegexOptions.CultureInvariant) == 0)
-            {
-                writer.WriteLine("TextInfo textInfo = CultureInfo.CurrentCulture.TextInfo;");
-                return true;
-            }
-
-            return false;
-        }
-
-        private static bool UseToLowerInvariant(bool hasTextInfo, RegexOptions options) => !hasTextInfo || (options & RegexOptions.CultureInvariant) != 0;
-
-        private static string ToLower(bool hasTextInfo, RegexOptions options, string expression) => UseToLowerInvariant(hasTextInfo, options) ? $"char.ToLowerInvariant({expression})" : $"textInfo.ToLower({expression})";
+        private static string ToLower(string expression) => $"textInfo.ToLower({expression})";
 
-        private static string ToLowerIfNeeded(bool hasTextInfo, RegexOptions options, string expression, bool toLower) => toLower ? ToLower(hasTextInfo, options, expression) : expression;
+        private static string ToLowerIfNeeded(string expression, bool toLower) => toLower ? ToLower(expression) : expression;
```

Comment:
Hopefully after the previous suggested changes these helpers can just be deleted.